Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -0,0 +1,860 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+// DolphinQt code copied and modified for Dolphin from the RPCS3 Qt utility for Creating, Loading
+// and Clearing skylanders
+
+#include "DolphinQt/SkylanderPortal/SkylanderPortalWindow.h"
+
+#include <QCheckBox>
+#include <QComboBox>
+#include <QCompleter>
+#include <QEvent>
+#include <QFileDialog>
+#include <QGroupBox>
+#include <QKeyEvent>
+#include <QKeySequence>
+#include <QLabel>
+#include <QMessageBox>
+#include <QPushButton>
+#include <QScrollArea>
+#include <QString>
+#include <QStringList>
+#include <QVBoxLayout>
+
+#include "Common/IOFile.h"
+
+#include "Core/Config/MainSettings.h"
+
+#include "DolphinQt/QtUtils/DolphinFileDialog.h"
+#include "DolphinQt/Settings.h"
+
+QString m_last_skylander_path;
+
+const std::map<const std::pair<const u16, const u16>, const std::string> list_skylanders = {
```

Comment:
```suggestion
const std::map<const std::pair<const u16, const u16>, const char*> list_skylanders = {
```
would make this map cheaper to construct and occupy less memory.